include/albatross/src/details/has_any_macros.hpp

```diff
@@ -15,29 +15,70 @@
 
 namespace albatross {
 
-#define HAS_METHOD(fname)                                                      \
-  template <typename T, typename... Args> class has_##fname {                  \
-    template <typename C, typename = decltype(std::declval<C>().fname(         \
-                              std::declval<Args>()...))>                       \
-    static std::true_type test(C *);                                           \
-    template <typename> static std::false_type test(...);                      \
-                                                                               \
-  public:                                                                      \
-    static constexpr bool value = decltype(test<T>(0))::value;                 \
-  };
+template <typename First, typename Second> struct TypePair {
```

Comment:
I think you could use `std::pair` as well. It should have `first_type` and `second_type` member types defined.

https://en.cppreference.com/w/cpp/utility/pair

Reply:
Yeah I can *almost* use it, but I found that it fails when you pass a `void` type as one of the types in a pair because the pair actually tries storing all the types.

Reply:
Oh! I guess that makes sense.